tests/trinity/core/p2p-proto/test_sync.py

```diff
@@ -80,20 +83,33 @@ def finalizer():
 
 
 @pytest.mark.asyncio
-async def test_regular_syncer(request, event_loop, chaindb_fresh, chaindb_20):
+async def test_regular_syncer(request, event_loop, event_bus, chaindb_fresh, chaindb_20):
+
     client_peer, server_peer = await get_directly_linked_peers(
         request, event_loop,
         alice_headerdb=FakeAsyncHeaderDB(chaindb_fresh.db),
         bob_headerdb=FakeAsyncHeaderDB(chaindb_20.db))
+
+    client_chain = FrontierTestChain(chaindb_fresh.db)
+    client_block_import_handler = BlockImportHandler(client_chain, event_bus)
+    client_block_import_api = EventBusBlockImporter(event_bus)
+
     client = RegularChainSyncer(
-        FrontierTestChain(chaindb_fresh.db),
+        client_chain,
         chaindb_fresh,
-        MockPeerPoolWithConnectedPeers([client_peer]))
+        MockPeerPoolWithConnectedPeers([client_peer]),
+        block_import_fn=client_block_import_api.coro_import_block)
```

Comment:
@carver @pipermerriam This test is a simulation of the idea that I described here: https://github.com/ethereum/py-evm/pull/1299#issuecomment-440648907

This offloads the `coro_import_block` call so that it doesn't run in the `networking` process (as it does when we merge  #1299) but in the process that hosts the `BlockImportHandler`. The idea is that this would run as an isolated plugin. A separate process just dedicated for block processing / validation.

The test doesn't do *the real thing* yet as there aren't multiple processes involved. But other than that, the setup isn't fake and the test works.

I'd like to turn this into something that can be enabled via feature flag (e.g. `--dedicated-chain-process`) so that we could essentially A/B test trinity and look at actual numbers.

What's kinda cool about this approach of using the event bus to delegate to the actual processor is that it enables loose coupling. Anyone can write a plugin that listens for these events and take whatever approach they want to boost up the processing. E.g. you could write a plugin that uses https://dask.org to scale out to many processes (or whatever other magic it uses)